memory/src/main/scala/filodb.memory/format/vectors/LongBinaryVector.scala

```diff
@@ -227,6 +228,21 @@ object LongVectorDataReader64 extends LongVectorDataReader {
     }
     if (element == item) first else first | 0x80000000
   }
+
+  final def changes(vector: BinaryVectorPtr, start: Int, end: Int): Double = {
+    require(start >= 0 && end < length(vector), s"($start, $end) is out of bounds, length=${length(vector)}")
+    val prev = Double.NaN
+    var addr = vector + OffsetData + start * 8
+    val untilAddr = vector + OffsetData + end * 8 + 8   // one past the end
+    var changes = 0
+    while (addr < untilAddr) {
+      val nextDbl = UnsafeUtils.getLong(addr)
+      // There are many possible values of NaN.  Use a function to ignore them reliably.
+      if (!java.lang.Double.isNaN(nextDbl) && prev != nextDbl) changes += 1
```

Comment:
It makes no sense to compare NaN for Long and DeltaDelta vectors, as NaN is not possible for integer values.

Reply:
Removed NaN checks for LongBinaryVectors